public/app/features/dashboard-scene/scene/types.ts

```diff
@@ -1,5 +1,85 @@
-import { BusEventWithPayload } from '@grafana/data';
-import { SceneObject } from '@grafana/scenes';
+import { BusEventWithPayload, RegistryItem } from '@grafana/data';
+import { SceneObject, VizPanel } from '@grafana/scenes';
+
+/**
+ * A scene object that usually wraps an underlying layout
+ * Dealing with all the state management and editing of the layout
+ */
+export interface DashboardLayoutManager extends SceneObject {
+  /**
+   * Notify the layout manager that the edit mode has changed
+   * @param isEditing
+   */
+  editModeChanged(isEditing: boolean): void;
+  /**
+   * We should be able to figure out how to add the explore panel in a way that leaves the
+   * initialSaveModel clean from it so we can leverage the default discard changes logic.
+   * Then we can get rid of this.
+   */
+  cleanUpStateFromExplore?(): void;
+  /**
+   * Not sure we will need this in the long run, we should be able to handle this inside internally
+   */
+  getNextPanelId(): number;
+  /**
+   * Remove an elemenet / panel
+   * @param element
+   */
+  removePanel(panel: VizPanel): void;
+  /**
+   * Creates a copy of an existing element and adds it to the layout
+   * @param element
+   */
+  duplicatePanel(panel: VizPanel): void;
+  /**
+   * Adds a new panel to the layout
+   */
+  addPanel(panel: VizPanel): void;
+  /**
+   * Add row
+   */
+  addNewRow(): void;
+  /**
+   * getVizPanels
+   */
+  getVizPanels(): VizPanel[];
```

Comment:
Something we can consider _later on_ is change the naming from `panel` to `element`. I suppose with more development we will allow adding other things than panels to the layouts (i.e. button in canvas, or tab in tabbed layout).